### PR TITLE
[MSYS-1040] Fix specs for chef-15 support

### PR DESCRIPTION
**Pull request by NAshwini**
Signed-off-by: NAshwini <ashwini.nehate@msystechnologies.com>

### Description

Fix specs for chef-15 support as per changes done by PR: https://github.com/chef/knife-cloud/pull/117

### CODE REVIEW

File: spec/unit/server_create_command_spec.rb [hunk 1]
```diff
@@ -110,89 +108,13 @@ class ServerCreate < Chef::Knife::Cloud::ServerCreateCommand
     include Chef::Knife::Cloud::ServerCreateOptions
   end
 
-  describe "#bootstrap options" do
-
-    it "set chef config knife options" do
```
Comment (Vasu1105) [inline spec/unit/server_create_command_spec.rb]: why we removed this one ?

File: spec/unit/bootstrap_protocol_spec.rb [hunk 2]
```diff
@@ -21,7 +21,7 @@
 
 describe Chef::Knife::Cloud::BootstrapProtocol do
   before do
-    @config = { bootstrap_protocol: "ssh" }
+    @config = { connection_protocol: "ssh" }
     @instance = Chef::Knife::Cloud::BootstrapProtocol.new(@config)
   end
```
Comment (btm) [inline spec/unit/bootstrap_protocol_spec.rb]: these before blocks could all be converted to `let`
```
let(:config) { { connection_protocol: "ssh" } }
let(:instance) { Chef::Knife::Cloud::BootstrapProtocol.new(@config) }
```

File: spec/unit/bootstrapper_spec.rb [hunk 3]
```diff
@@ -40,9 +40,9 @@
     it "executes with correct method calls" do
       @ssh_bootstrap_protocol = Chef::Knife::Cloud::SshBootstrapProtocol.new(@config)
       allow(@instance).to receive(:create_bootstrap_protocol).and_return(@ssh_bootstrap_protocol)
-      @unix_distribution = Chef::Knife::Cloud::UnixDistribution.new(@config)
+      @bootstrapdistribution = Chef::Knife::Cloud::BootstrapDistribution.new(@config)
```
[the remark this answers is not in the document]
Comment (btm) [reply spec/unit/bootstrapper_spec.rb]: Agree we should switch to camel case.

Also most of these should be local variables not instance variables.

File: spec/unit/bootstrapper_spec.rb [hunk 4]
```diff
@@ -66,13 +66,8 @@
...
       end
 
-      it "instantiates Windows Distribution class." do
-        expect(Chef::Knife::Cloud::WindowsDistribution).to receive(:new).with(@config)
-        @instance.create_bootstrap_distribution
-      end
-
-      it "doesn't instantiate Unix Distribution class." do
-        expect(Chef::Knife::Cloud::UnixDistribution).to_not receive(:new)
+      it "instantiates Distribution class." do
```
Comment (btm) [inline spec/unit/bootstrapper_spec.rb]: nit: the period can be dropped here.

`it "instantiates the BootstrapDistribution class" do`

File: spec/unit/server_create_command_spec.rb [hunk 5]
```diff
@@ -84,7 +82,7 @@
       instance.after_exec_command
     end
 
-    # Currently the RangeError is occured when image_os_type is set to linux and bootstrap-protocol is set to ssh before windows server bootstrap.
+    # Currently the RangeError is occured when image_os_type is set to linux and --connection-protocol is set to ssh before windows server bootstrap.
```
Comment (btm) [inline spec/unit/server_create_command_spec.rb]: > The RangeError is *raised...

File: spec/unit/server_create_command_spec.rb [hunk 6]
```diff
@@ -110,89 +108,13 @@ class ServerCreate < Chef::Knife::Cloud::ServerCreateCommand
     include Chef::Knife::Cloud::ServerCreateOptions
   end
 
-  describe "#bootstrap options" do
-
-    it "set chef config knife options" do
+  describe "Bootstrap Protocol option" do
+    it "not to be set in chef config knife options" do
```
Comment (btm) [inline spec/unit/server_create_command_spec.rb]: can the describe and it line descriptions be improved here, I'm not really sure what this test is supposed to be doing.

File: spec/unit/server_create_command_spec.rb [hunk 7]
```diff
@@ -110,89 +108,13 @@ class ServerCreate < Chef::Knife::Cloud::ServerCreateCommand
...
-    end
-  end
-
-  describe "#before_bootstrap" do
-
-    before(:all) { @instance = ServerCreate.new }
-
-    context "bootstrap_protocol shh" do
-      before { @instance.config[:bootstrap_protocol] = "ssh" }
+      bootstrap_protocol = "bootstrap_protocol"
```
Comment (btm) [inline spec/unit/server_create_command_spec.rb]: I'd probably change this to something like `bootstrap_protocol = "nothing"` or such. it's slightly confusing having the value the same as the variable name.

File: spec/unit/server_create_command_spec.rb (hunk 1 above)
Comment (NAshwini) [reply spec/unit/server_create_command_spec.rb]: As we have inherited **bootstrap_url, bootstrap_install_command, bootstrap_wget_options, bootstrap_curl_options,     bootstrap_no_proxy**  options from chef core and removed it from this commit changes https://github.com/chef/knife-cloud/pull/117/files#diff-3503536858f753f9066eca592720490eL185 . we need to removed this part of code.

File: spec/unit/server_create_command_spec.rb (hunk 6 above)
Comment (NAshwini) [reply spec/unit/server_create_command_spec.rb]: It is just to check by any chance we are not sending `--bootstrap-protocol` option to @bootstarp object.

**Comment by NAshwini** [comment]
Thanks @btm and @Vasu1105. I have fixed the specs as per review comments.

**Pull request merged**